test/e2e_federation/upgrades/generic.go

```diff
@@ -58,8 +59,13 @@ func (t *GenericUpgradeTest) Teardown(f *fedframework.Framework) {
 	// Rely on the namespace deletion to clean up everything
 }
 
-// SecretUpgradeTest validates that a secret remains propagated before
-// and after a controlplane upgrade
-func SecretUpgradeTest() Test {
-	return &GenericUpgradeTest{adapter: &typeadapters.SecretAdapter{}}
+func GenericUpgradeTests() []Test {
```

Comment:
I like comments, but I think they only have value where intent can't be inferred from code.  Otherwise they're a maintenance burden.  I think for trivial functions such as this - all it does is collect upgrade tests for compatible types - there is minimal benefit from providing a comment.